contrib/format-esri/src/main/java/org/apache/drill/exec/store/esri/ShpBatchReader.java

```diff
@@ -180,6 +182,10 @@ private String byteArrayToString(byte[] in) {
   }
 
   private void processShapefileSet(RowSetLoader rowWriter, final int gid, final Geometry geom, final Object[] dbfRow) {
+    if (maxRecords > 0 && rowWriter.rowCount() >= maxRecords) {
```

Comment:
Same code happens to be used in multiple places? Maybe static function will make it clearer

Reply:
Fixed in https://github.com/apache/drill/pull/2092/commits/90c82f22cf201c046964a8a8ca45d8d0f66b2168